--- a/src/modules/battery.cpp
+++ b/src/modules/battery.cpp
@@ -65,7 +65,7 @@
     }
 
     m_rate_reader = make_unique<rate_reader>([this] {
-      unsigned long rate{std::strtoul(file_util::contents(m_frate).c_str(), nullptr, 10)};
+      unsigned long rate{std::abs(std::strtol(file_util::contents(m_frate).c_str(), nullptr, 10))};
       unsigned long volt{std::strtoul(file_util::contents(m_fvoltage).c_str(), nullptr, 10) / 1000UL};
       unsigned long now{std::strtoul(file_util::contents(m_fcapnow).c_str(), nullptr, 10)};
       unsigned long max{std::strtoul(file_util::contents(m_fcapfull).c_str(), nullptr, 10)};
